### ISSUE TITLE
Minimum version of Pydantic needs to be updated in `setup.py` to allow UI to work properly

### ISSUE THREAD

**Issue by arosen93** [AgnostiqHQ/covalent#1765]
### Environment

- **Covalent version**: 0.227.0-rc0
- **Python version**: 3.10.12
- **Operating system**: Ubuntu

### What is happening?

The UI is not showing the status of launched jobs.

![image](https://github.com/AgnostiqHQ/covalent/assets/8674072/d9b0b902-3380-406d-b836-6f5d76aad72c)

### How can we reproduce the issue?

Run an example Covalent script. I used my typical example below.

```python
import covalent as ct

@ct.electron
def add(a, b):
	return a+b

@ct.electron
def mult(a, b):
	return a*b

@ct.electron
def make_more(a):
	return [a]*3

@ct.electron
@ct.lattice
def add_distributed(vals):
	return [add(val,2) for val in vals]

@ct.lattice
def workflow(a, b, c):
	output1 = add(a,b)
	output2 = mult(output1,c)
	output3 = make_more(output2)
	output4 = add_distributed(output3)
	return output4

dispatch_id=ct.dispatch(workflow)(1,2,3)
ct.get_result(dispatch_id,wait=True)
```

### What should happen?

The UI should show the results.

### Any suggestions?

No :(

I tried completely removing my entire Conda env, removing the following files, and a few other things without luck. Reverting to 0.226.0-rc0 resolves the issue for me.

Directories removed: 
- `~/.cache/covalent`
- `~/.config/covalent`
- `~/.local/.share/covalent`

Strangely, just before this error I had a separate bug where the function names and descriptions for sublattice electrons were showing up under the wrong electrons. But I'll see if I can reproduce that error on >=0.227.0-rc0 once the UI is working.

**Comment by santoshkumarradha** [comment]
@Prasy12 could you pick this up as high P bug please ?

**Comment by Aravind-psiog** [comment]
@arosen93 , We are looking into this. Can you share the log file `.cache/covalent/covalent_ui.log`. It would help us debug the issue. TIA

**Comment by Aravind-psiog** [comment]
### Environment
**Covalent version**: 0.227.0-rc0
**Python version**: 3.10.12
**Operating system**: Ubuntu 22.04.2 LTS

These are the steps we followed:

1. Created a new conda with Python 3.10.12
2. Installed covalent with `pip install covalent==0.227.0-rc0`
3. purged covalent files by `covalent purge --hard -y`
4. started covalent
5. Dispatched a few workflows
6. Dispatched the above example workflow

We were able to see the UI working as expected

![os-stat01](https://github.com/AgnostiqHQ/covalent/assets/100823292/04569735-4dcc-4ac1-b6a2-2e66d3110970)
------------------------------------------------------------------------------------------------------------------------------------------------------------
![os-stat02](https://github.com/AgnostiqHQ/covalent/assets/100823292/efcd062f-36dc-4dbf-b092-ad80601b7419)

Please let us know if we are missing something. If the issue persists, the log file would be of great help for us to debug and resolve the issue. Thanks @arosen93

**Comment by arosen93** [comment]
@Aravind-psiog: Thank you for the detailed report. It was very helpful in narrowing this down!

In short, the issue is related to Pydantic, and we need to update the minimum supported version in `setup.py`, as I justify below.

When I do a clean install of Covalent as you laid out, everything works as expected. Beautiful.

When I do an install of Covalent with a [package I'm developing](https://github.com/Quantum-Accelerators/quacc), I get the error I reported before. This led me to believe it was a dependency issue.

I then opened the `covalent_ui.log` you mentioned. Thank you for letting me know about this file, as this was extremely helpful as well. The log file indicated that it is an issue related to Pydantic.

```python
  File "pydantic/main.py", line 574, in pydantic.main.BaseModel.from_orm
pydantic.errors.ConfigError: You must have the config attribute orm_mode=True to use from_orm
```

In my environment, I had been using Pydantic 1.10.12, which is "supported" based on Covalent's `requirements.txt` that lists `pydantic>=1.10.1`. However, when you install only Covalent in a clean conda env, you get Pydantic 2.3.0.

I installed only Covalent only in a clean conda env _but with Pydantic 1.10.12_, I can reproduce the error in the UI.

**Summary**: The Pydantic version in `setup.py`/`requirements.txt` needs to be updated from `pydantic>=1.10.1` to... something else, likely `pydantic>2`.
